### PR TITLE
[RFC] Small revisions to msgpack and plugin documents

### PR DESCRIPTION
**Pull request by alysbrooks**
I've made some small line-edits (i.e., no rearranging sentences or paragraphs) to msgpack_rpc.txt and remote_plugin.txt.

I've separated out capitalization and strict grammar fixes (first commit) from more subjective rewordings to make sentences flow better (second commit).

### CODE REVIEW

File: runtime/doc/msgpack_rpc.txt [hunk 1]
```diff
@@ -17,22 +17,22 @@ The Msgpack-RPC Interface to Nvim				  *msgpack-rpc*
...
+The primary way to control a running Nvim instance is through
 MessagePack-RPC, a messaging protocol that uses the MessagePack serialization
 format: https://github.com/msgpack/msgpack/blob/7498cf3/spec.md.
-From now on, we'll be referring to the protocol as msgpack-rpc.
+From now on, we refer to the protocol as msgpack-rpc.
 
 At this point, only plugins use msgpack-rpc, but eventually even user
-interaction will be achieved through the protocol, since user interfaces will
-be separate programs that control a headless Nvim instance.
+interaction will happen through the protocol, since user interfaces will be
```
Comment (Pyrohh) [inline runtime/doc/msgpack_rpc.txt]: maybe `... even user interaction will happen through it, ... `? "the protocol" seems redundant since it's just been mentioned.

**Comment by Pyrohh** [comment]
I think it's fine to squash these commits together. I'm usually for complex pull requests being broken up into a logical progression of commits, but this is fairly straightforward.

Besides that, please remove the trailing whitespace you've introduced (`git log --check` will show where it is).

File: runtime/doc/msgpack_rpc.txt [hunk 2]
```diff
@@ -45,19 +45,19 @@ non-static API functions, converting/validating arguments and return values
...
-   dynamically-typed languages, which can define functions at runtime.
+1. By connecting to a running Nvim instance and calling `vim_get_api_info`
+   via msgpack-rpc. This is best for clients written in dynamically-typed 
+   languages, which can define functions at runtime.
 
-2. Through the `--api-info` command-line option, which makes nvim dump a
-   msgpack blob containing metadata to stdout and exit. This is preferred
-   when writing clients for statically-typed languages, which require a
-   separate compilation step.
+2. By running Nvim with the `--api-info` command-line option, which makes Nvim
```
Comment (Pyrohh) [inline runtime/doc/msgpack_rpc.txt]: `running` -> `starting`?

File: runtime/doc/msgpack_rpc.txt [hunk 3]
```diff
@@ -45,19 +45,19 @@ non-static API functions, converting/validating arguments and return values
...
+   dump a msgpack blob containing metadata to stdout and exit. This is
```
Comment (Pyrohh) [inline runtime/doc/msgpack_rpc.txt]: maybe `..., which make Nvim dump msgpack metadata to standard output and exit.`?

File: runtime/doc/msgpack_rpc.txt [hunk 4]
```diff
@@ -45,19 +45,19 @@ non-static API functions, converting/validating arguments and return values
...
+   via msgpack-rpc. This is best for clients written in dynamically-typed 
+   languages, which can define functions at runtime.
 
-2. Through the `--api-info` command-line option, which makes nvim dump a
-   msgpack blob containing metadata to stdout and exit. This is preferred
-   when writing clients for statically-typed languages, which require a
-   separate compilation step.
+2. By running Nvim with the `--api-info` command-line option, which makes Nvim
+   dump a msgpack blob containing metadata to stdout and exit. This is 
+   best clients in statically-typed languages, which require a separate
```
Comment (Pyrohh) [inline runtime/doc/msgpack_rpc.txt]: `best clients` -> `best for clients`

File: runtime/doc/msgpack_rpc.txt [hunk 5]
```diff
@@ -67,31 +67,31 @@ Python and the `pyyaml`/`msgpack-python` pip packages):
...
+1. Through Nvim's stdin/stdout when it's started with the `--embed` option. 
+   This is how other programs can embed Nvim.
 
 2. Through the stdin/stdout of a program spawned by the |rpcstart()| function.
 
 							*$NVIM_LISTEN_ADDRESS*
 3. Through the socket automatically created with each instance. To find out
-   the socket location (which is random by default) from a running nvim
+   the socket location (which is random by default) for a running Nvim
    instance, one can inspect the |$NVIM_LISTEN_ADDRESS| environment variable:
```
Comment (Pyrohh) [inline runtime/doc/msgpack_rpc.txt]: `one can inspect the` -> `see the` ?

File: runtime/doc/msgpack_rpc.txt [hunk 6]
```diff
@@ -67,31 +67,31 @@ Python and the `pyyaml`/`msgpack-python` pip packages):
...
+   |$NVIM_LISTEN_ADDRESS| environment variable in a shell before starting Nvim:
 >
     NVIM_LISTEN_ADDRESS=127.0.0.1:6666 nvim
 <
 Connecting to the socket is the easiest way a programmer can test the API,
 which can be done through any msgpack-rpc client library or fully-featured
-Nvim client (which we'll see below). Here's a ruby script that will print the
-string 'hello world!' on the current nvim instance:
+Nvim client (which we'll see below). Here's a Ruby script that prints 'hello 
+world!' on the current Nvim instance:
```
Comment (Pyrohh) [inline runtime/doc/msgpack_rpc.txt]: perhaps `in the current` would be more suitable here

File: runtime/doc/msgpack_rpc.txt [hunk 7]
```diff
@@ -189,10 +188,10 @@ An API method expecting one of these types may be passed an integer instead,
 although they are not interchangeable. For example, a Buffer may be passed as
 an integer, but not a Window or Tabpage.
 
-The most reliable way of determining the type codes for the special nvim types
-is at runtime by inspecting the `types` key of metadata dictionary returned by
-`vim_get_api_info` method. Here's an example json representation of the
-`types` object:
+The most reliable way of determining the type codes for the special Nvim types
+is to inspect the `types` key of metadata dictionary returned by the
+`vim_get_api_info` method at runtime. Here's an example json representation of
```
Comment (Pyrohh) [inline runtime/doc/msgpack_rpc.txt]: `json` -> `JSON`

File: runtime/doc/remote_plugin.txt [hunk 8]
```diff
@@ -15,34 +15,34 @@ Nvim support for remote plugins 		  *remote-plugin*
...
 plugins, coprocesses that have a direct communication channel (via
 |msgpack-rpc|) with the Nvim process.
 
-Even though these plugins are running in separate processes they can call, be
-called, and receive events just as if the code was being executed in the main
+Even though these plugins run in separate processes they can call, be called,
+and receive events just as if the plugin's code were executed in the main
 process.
 
+
```
Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: An extra line such as this doesn't look to be present anywhere else in this document, so should probably be removed.

File: runtime/doc/remote_plugin.txt [hunk 9]
```diff
@@ -15,34 +15,34 @@ Nvim support for remote plugins 		  *remote-plugin*
...
 ==============================================================================
 2. Plugin hosts					    *remote-plugin-hosts*
 
 While plugins can be implemented as arbitrary programs that communicate
 directly with the high-level Nvim API and are called via |rpcrequest()| and
-|rpcnotify()|, that is not the best approach available. Instead, developers
-should first check if a plugin host implementation is available for their
-chosen programming language.
+|rpcnotify()|, that is not the best approach. Instead, developers should first
+check whether a plugin host is available for their chosen programming language.
```
Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: Why was `implementation` removed?

File: runtime/doc/remote_plugin.txt [hunk 10]
```diff
@@ -82,16 +82,16 @@ of requests made to it. Here's the plugin source code:
 <
 
 As can be seen, the plugin is implemented using pure Python idioms (classes,
```
Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: maybe `... using idiomatic Python (classes, ...`

File: runtime/doc/remote_plugin.txt [hunk 11]
```diff
@@ -82,16 +82,16 @@ of requests made to it. Here's the plugin source code:
...
-methods, and decorators), the translation between these language-specific
-idioms to vimscript occurs while the plugin manifest is being generated (see
+methods, and decorators). The translation between these language-specific
+idioms to Vimscript occurs while the plugin manifest is being generated (see
 below).
```
Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: `(see below)` -> `(see the next section)`

File: runtime/doc/remote_plugin.txt [hunk 12]
```diff
@@ -101,35 +101,34 @@ the remote plugin manifest must be generated with `:UpdateRemotePlugins`.
 4. Remote plugin manifest			    *remote-plugin-manifest*
 
 Just installing remote plugins to "rplugin/{host}" isn't enough for them to be
-automatically loaded when required. The `:UpdateRemotePlugins` command must be
-executed every time a remote plugin is installed, updated, or deleted.
+automatically loaded when required. You must execute `:UpdateRemotePlugins`
+every time a remote plugin is installed, updated, or deleted.
```
Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: I prefer the old wording due to the use of active voice.

File: runtime/doc/remote_plugin.txt [hunk 13]
```diff
@@ -101,35 +101,34 @@ the remote plugin manifest must be generated with `:UpdateRemotePlugins`.
...
 (commands/autocommands/functions) defined by all remote plugins, with each
-entity associated with the host and plugin path. The manifest can be seen as a
-generated extension to the user's vimrc (it even has the vimrc filename
-prepended).
+entity associated with the host and plugin path. The manifest is a generated
+extension to the user's vimrc (it even has the vimrc filename prepended).
 
-The manifest declarations are nothing but calls to the remote#host#RegisterPlugin
-function, which will take care of bootstrapping the host as soon as the
+Manifest declarations are nothing but calls to the remote#host#RegisterPlugin
```
Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: `nothing more than`?

Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: Also, `remote#host#RegisterPlugin` should probably be surrounded by backtics.

File: runtime/doc/remote_plugin.txt [hunk 14]
```diff
@@ -101,35 +101,34 @@ the remote plugin manifest must be generated with `:UpdateRemotePlugins`.
...
-with the example, say the java plugin is a semantic completion engine for java
-source files. If it defines the autocommand "BufEnter *.java", then the java
-host will only be spawned when files ending with ".java" are loaded.
-
-If the explicit call to `:UpdateRemotePlugins` seems incovenient, try to see
-it like this: It's a way to give IDE-like capabilities to nvim while still
-keeping it fast and lightweight for general use. It can also be seen as
-analogous to the |:helptags| facility.
+with the example, say the Java plugin is a semantic completion engine for Java
+source files. If it defines the autocommand "BufEnter *.java", then the Java
```
Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: `source files` -> `source code` -- I'd imagine completion plugins work fine if the file hasn't been saved to disk yet.

Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: or just `Java code`, as the meaning in this context is obvious IMO.

File: runtime/doc/remote_plugin.txt [hunk 15]
```diff
@@ -101,35 +101,34 @@ the remote plugin manifest must be generated with `:UpdateRemotePlugins`.
...
+host is spawned only when files ending with ".java" are loaded.
```
Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: Maybe `... spawned only for buffers matching "*.java".` I think it's reasonable to assume the reader is aware of such wildcards.

File: runtime/doc/remote_plugin.txt [hunk 16]
```diff
@@ -101,35 +101,34 @@ the remote plugin manifest must be generated with `:UpdateRemotePlugins`.
...
+
+If the explicit call to `:UpdateRemotePlugins` seems incovenient, try to see it
+like this: It's a way to give IDE-like capabilities to Nvim while still keeping
+it fast and lightweight for general use. It's also analogous to the |:helptags|
+facility.
```
Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: `facility` -> `command` ?

File: runtime/doc/remote_plugin.txt [hunk 17]
```diff
@@ -101,35 +101,34 @@ the remote plugin manifest must be generated with `:UpdateRemotePlugins`.
...
-If the explicit call to `:UpdateRemotePlugins` seems incovenient, try to see
-it like this: It's a way to give IDE-like capabilities to nvim while still
-keeping it fast and lightweight for general use. It can also be seen as
-analogous to the |:helptags| facility.
+with the example, say the Java plugin is a semantic completion engine for Java
+source files. If it defines the autocommand "BufEnter *.java", then the Java
+host is spawned only when files ending with ".java" are loaded.
+
+If the explicit call to `:UpdateRemotePlugins` seems incovenient, try to see it
+like this: It's a way to give IDE-like capabilities to Nvim while still keeping
```
Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: maybe "It's a way to **provide** IDE-like capabilities **in** Nvim".

**Comment by Pyrohh** [comment]
Besides the comments this looks pretty good, thanks for doing this.

File: runtime/doc/msgpack_rpc.txt (hunk 2 above)
Comment (nhynes) [inline runtime/doc/msgpack_rpc.txt]: This sentence is missing an independent clause. Maybe reword to "Running Nvim with the `--api-info` command-line option makes Nvim..."

File: runtime/doc/remote_plugin.txt (hunk 12 above)
Comment (alecbrooks) [inline runtime/doc/remote_plugin.txt]: The new wording is active. 'You' is the subject, and it's doing the acting.

Or do you mean that the new wording's imperative?

File: runtime/doc/remote_plugin.txt (hunk 9 above)
Comment (alecbrooks) [inline runtime/doc/remote_plugin.txt]: It didn't seem to be adding anything. The rest of the file refers to "plugin hosts," and it sounded fine that way, so I got rid of "implementation" here.

File: runtime/doc/msgpack_rpc.txt [hunk 18]
```diff
@@ -45,19 +45,19 @@ non-static API functions, converting/validating arguments and return values
...
+   via msgpack-rpc. This is best for clients written in dynamically-typed
+   languages, which can define functions at runtime.
 
-2. Through the `--api-info` command-line option, which makes nvim dump a
-   msgpack blob containing metadata to stdout and exit. This is preferred
-   when writing clients for statically-typed languages, which require a
-   separate compilation step.
+2. By starting Nvim with the `--api-info` command-line option, which makes Nvim
+   dump a msgpack metadata to standard output and exit. This is best for clients
+   written in statically-typed languages, which require a separate compilation
```
Comment (alecbrooks) [inline runtime/doc/msgpack_rpc.txt]: Should this be "which *often* require a separate compilation step"?

I'm not actually sure about why a separate compilation step matters. Should it say that functions can't be defined after compile time?

**Comment by alecbrooks** [comment]
I made almost all of the suggested changes and left some comments explaining the rest. Thanks for reviewing, everyone!

File: runtime/doc/remote_plugin.txt (hunk 12 above)
Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: > Or do you mean that the new wording's imperative?

Heh, yes I meant that. Sorry for the confusion.

File: runtime/doc/remote_plugin.txt (hunk 9 above)
Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: I suppose you're right.

File: runtime/doc/msgpack_rpc.txt (hunk 18 above)
Comment (Pyrohh) [inline runtime/doc/msgpack_rpc.txt]: Regarding the actual documentation above, I don't think `dump a msgpack metadata` makes much sense, kind of like `I'll give you a data containing my address`. I might have been unclear in my initial request; anyways, how aboutabout `dump a blob of msgpack ...`, or something like that.

> Should this be "which often require a separate compilation step"?

I'd think they would always require a separate compilation step, as type checking of a function can't be done at compile time if the function doesn't exist.



> The first sentences of 1. and 2. still lack independent clauses. I think that's okay for a list, as long as it sounds natural.

Yes I think so too. I prefer the slightly informal nature of it; it sounds more natural, as you say.

**Comment by Pyrohh** [comment]
Beside the nit I [mentioned here](https://github.com/neovim/neovim/pull/2831#discussion_r32331718), this LGTM.

**Comment by Pyrohh** [comment]
Oops, and https://github.com/neovim/neovim/pull/2831#discussion_r32492186 as well.

File: runtime/doc/remote_plugin.txt (hunk 12 above)
Comment (alecbrooks) [inline runtime/doc/remote_plugin.txt]: Existing Vim help files use "you" pretty heavily. What don't you like about it?

File: runtime/doc/msgpack_rpc.txt (hunk 18 above)
Comment (alecbrooks) [inline runtime/doc/msgpack_rpc.txt]: What if I just remove the "a" (`which makes Nvim dump a msgpack metadata to standard output and exit.`)?

File: runtime/doc/remote_plugin.txt (hunk 12 above)
Comment (Pyrohh) [inline runtime/doc/remote_plugin.txt]: I like how my suggestion assumes you know what to do with the command, as opposed to explicitly saying `You must execute ...`.

Anyways, I don't want to block this PR with what's arguably pretty subjective, so please feel free to leave it as is.

File: runtime/doc/msgpack_rpc.txt (hunk 18 above)
Comment (Pyrohh) [inline runtime/doc/msgpack_rpc.txt]: > What if I just remove the "a"

Sure.

File: runtime/doc/remote_plugin.txt (hunk 12 above)
Comment (alecbrooks) [inline runtime/doc/remote_plugin.txt]: I'd like to err on the side of active voice and shorter wording, so I'll leave as-is.

**Comment by alecbrooks** [comment]
Both of Pyrohh's comments are addressed (one with a reply instead of a change), so it should be ready to merge.

**Comment by Pyrohh** [comment]
Would you like to take a look @justinmk @fwalch? I'll merge it if there's no more reviews.

**Comment by Pyrohh** [comment]
Thanks alec, merged with a few small improvements I didn't notice before.
